File: 1151.cpp
#include <bits/stdc++.h>

using namespace std;

int main (){

    freopen("input.txt", "r", stdin);
    freopen("output.txt", "w", stdout);

    int x, x1 = 0, x2 = 1, x3 = 0;
    
    cin >> x;

    if (x == 0 or x == 1){
        printf("%d", x1);
        return 0;
    }

 
    printf("%d ", x1);

    for(int i = 2; i <= x; i++){

        if(i < x){
                x1 = x2;
                x2 = x3;
                x3 = x1 + x2;
                printf("%d ", x3);
        }
        if(i == x){
            printf("%d\n", x2 + x3);
            break;
        }
    }
  

}

File: 1151.py
import sys
sys.stdin = open('input.txt', 'r')
sys.stdout = open('output.txt', 'w')

x = int(input())
v = []

    

for i in range (0,x):

    if(x == 0 or x == 1):
        op = x
        v.append(int(op))
        break
    if(i == 0):
        op = 0
    if(i > 0):
        if(i == 1):
            op = 1
        else:
            op = v[i-2]+v[i-1]
    

    if(i == x):
        break
    v.append(int(op))

print(v)

